### PR TITLE
Open Movie Database Integrations

### PR DESCRIPTION
**Pull request by jcalfeeSFDC**
Contains class to connect to, and parse, data on the Open Movie Database. Wrapper class to hold the data is included. Unit tests for the connection and wrapper class are included. I am beginning to make the Aura and LWC components to allow the users to enter data automatically.

### CODE REVIEW

File: force-app/main/default/classes/OmdbConnection.cls [hunk 1]
```diff
@@ -0,0 +1,138 @@
+global with sharing class OmdbConnection {
+    private String httpMethod = 'GET';
+    private String omdbURLBase = 'http://www.omdbapi.com/';
+
+    // (?) TODO:    -> Reference from encrypted .txt file (?)
```
Comment (douglas-riceSFDC) [inline force-app/main/default/classes/OmdbConnection.cls:5]: You generally want to remove TODOs before you commit. They can be confusing for someone pulling down the code and not knowing if the TODOs are still relevant or not. They are okay to have in code while you are working, but once you finish a feature, you should remove them

File: force-app/main/default/classes/OmdbConnectionTest.cls [hunk 2]
```diff
@@ -0,0 +1,207 @@
+@isTest
+private class OmdbConnectionTest {
+    // TODO ADD SYSTEM ASSERTS
```
Comment (douglas-riceSFDC) [inline force-app/main/default/classes/OmdbConnectionTest.cls]: I would delete unused functions or large comment blocks very rarely do you need them and they clutter up a class and kind of stick around forever. With version control, we can bring them back if we need to so there isn't a reason to keep them commented out

**Review by douglas-riceSFDC — CHANGES_REQUESTED** [review]
Take a look at my comments that I've left. They aren't critical to implement, but still important for the overall health of our codebase. In addition, you should remove the .json files in the PR because if someone pulls the branch down, all of them will come with it and it might mess up someone's own personal VSCode.

File: force-app/main/default/classes/OmdbConnection.cls (hunk 1 above)
Comment (jcalfeeSFDC) [reply force-app/main/default/classes/OmdbConnection.cls:5]: Ok that's good to know thank you! Generally speaking, would it be best practice to keep the 'key' in a separate file? I'm sure it's fine for this project, but I can always change that if needed.

Comment (douglas-riceSFDC) [reply force-app/main/default/classes/OmdbConnection.cls:5]: It really depends... There are a number of ways that Salesforce allows for storing secrets and encrypted information each with pros and cons depending on the value you are trying to store and how secure you need to keep it.

In this case we probably would want to store the API key and URL in a Custom Metadata or Custom Setting within Salesforce. The data isn't really a secret per se, so we don't have to worry too much about keeping the value hidden like say a username or password. Storing the value in a Custom Metadata or Custom Setting would just allow for us to maintain the values without needing to update a class each time in the event it changes

**Review by douglas-riceSFDC — APPROVED** [review]

**Pull request merged**